File: src/leetcode/java/EliminationGame_390.java
package leetcode.java;

public class EliminationGame_390 {
  public int lastRemaining(int n) {
    boolean left = true;
    int step = 1, head = 1;
    int remaining = n;
    while (remaining > 1) {
      if (left || remaining % 2 == 1) head += step;
      left = !left;
      remaining /= 2;
      step *= 2;
    }
    return head;
  }
}

File: src/leetcode/python/EliminationGame_390.py
class EliminationGame_390(object):
    def lastRemaining(self, n):
        """
        :type n: int
        :rtype: int
        """
        left = True
        step, head = 1, 1
        remaining = n
        while remaining > 1:
            if left or remaining % 2 == 1:
                head += step
            left = False if left else True
            remaining /= 2
            step *= 2
        return head
